client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -586,6 +586,17 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
       oldEpoch: Int,
       oldPartition: PartitionLocation,
       cause: Option[StatusCode] = None): Unit = {
+    if (!registeredShuffle.contains(shuffleId)) {
```

Comment:
I think we should move these checks inside reallocateChangePartitionRequestSlotsFromCandidates, because after that point the ReserveSlots will be called.

Reply:
> I think we should move these checks inside reallocateChangePartitionRequestSlotsFromCandidates, because after that point the ReserveSlots will be called.

Then it can work for batch handling too. Nice suggestion, but how about the current? Moving this part inside `reallocateChangePartitionRequestSlotsFromCandidates` is a little strange, how about current?